src/shims/backtrace.rs

```diff
@@ -141,10 +189,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
 
-        this.write_immediate(name_alloc.to_ref(this), &this.mplace_field(&dest, 0)?.into())?;
-        this.write_immediate(filename_alloc.to_ref(this), &this.mplace_field(&dest, 1)?.into())?;
-        this.write_scalar(lineno_alloc, &this.mplace_field(&dest, 2)?.into())?;
-        this.write_scalar(colno_alloc, &this.mplace_field(&dest, 3)?.into())?;
+        match flags {
+            0 => {
+                // These are "mutable" allocations as we consider them to be owned by the callee.
+                let name_alloc =
+                    this.allocate_str(&name, MiriMemoryKind::Rust.into(), Mutability::Mut);
+                let filename_alloc =
+                    this.allocate_str(&filename, MiriMemoryKind::Rust.into(), Mutability::Mut);
+
+                this.write_immediate(
+                    name_alloc.to_ref(this),
+                    &this.mplace_field(&dest, 0)?.into(),
+                )?;
+                this.write_immediate(
+                    filename_alloc.to_ref(this),
+                    &this.mplace_field(&dest, 1)?.into(),
+                )?;
+            }
+            1 => {
+                this.write_scalar(
+                    Scalar::from_machine_usize(name.len().try_into().unwrap(), this),
+                    &this.mplace_field(&dest, 0)?.into(),
+                )?;
+                this.write_scalar(
+                    Scalar::from_machine_usize(filename.len().try_into().unwrap(), this),
+                    &this.mplace_field(&dest, 1)?.into(),
+                )?;
+            }
+            _ => unreachable!("should have already errored on invalid flags"),
```

Comment:
Up in `handle_miri_get_backtrace` this is where you do `throw_unsup_format`; seems reasonable to use the same structure here instead of an early flag check.